### ISSUE TITLE
Capturing stderr

### ISSUE THREAD

**Issue by mgunyho** [rust-cli/rexpect#108]
Hi, I'm trying to write integration tests for an interactive CLI application (full-screen, like vim). Basically in my test, I want to run the main app binary, send some keystrokes to it, and then check the stdout and stderr. `rexpect` seems like a good fit, and it works for stdout, but I can't seem to find a way to read the stderr. Is there a way to read it?

**Comment by matthiasbeyer** [comment]
Huh, to be honest I don't think there is one.

You might be able to hack something with the `NBReader` and use that on an stderr stream, but from what I understand you want to read both stdout and stderr from one session and I don't think that is possible yet.

@philippkeller @petreeftime or am I missing something here?

**Comment by mgunyho** [comment]
> from what I understand you want to read both stdout and stderr from one session

Yes, ideally this is what I would like to do. I suppose in the general case this might be tricky to handle because messages can arrive in each concurrently, but for my case I just need a "one-shot" approach, where I send a bunch of key commands and then inspect the full output of both stdout and stderr.

**Comment by mgunyho** [comment]
Hmm, actually it looks like both stdout and stderr are read by `NBReader`:
```rust
use rexpect;
fn main() {
    let mut p = rexpect::session::spawn(
        // 'hello' goes to stdout, the exception backtrace to stderr
        "python3 -c \"print('hello'); raise Exception\"",
        Some(10_000)
    ).unwrap();

    println!("output:\n{}", p.exp_eof().unwrap());
}
```
This prints both 'hello' and the backtrace.

**Comment by mgunyho** [comment]
After searching around a bit more, it seems normal that the stdout and stderr for a pty are mixed, see for example here: https://stackoverflow.com/questions/16838624/why-does-openssh-requesttty-cause-stderr-redirected-to-stdout. I guess that makes sense: the user of my application sitting in front of a terminal would see the stdout and stderr mixed anyway. So I think I'll just write tests with that in mind.
